internals/daemon/api_notices.go

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) 2023 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package daemon
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"regexp"
+	"time"
+
+	"github.com/canonical/pebble/internals/overlord/state"
+)
+
+// A very loose regex to ensure client keys are in the form "domain.com/key"
+var clientKeyRegexp = regexp.MustCompile(`([a-z0-9-_]+\.)+[a-z0-9-_]+/.+`)
```

Comment:
Fair enough. I've removed underscore from allowed domain chars (it's extremely rare and often disallowed anyway), and I tightened up the allowed chars in the key part -- though not quite as tight as you suggested. I've made it `[A-Za-z0-9./-]+` so that things like `/Foo-Bar/index.html` are allowed.

Oh, and I noticed I was missing the `^` and `$` (start and end) anchors! So I added those.